compiler/rustc_hir_analysis/src/check/wfcheck.rs

```diff
@@ -136,6 +138,8 @@ where
         infcx.implied_bounds_tys_compat(param_env, body_def_id, &assumed_wf_types, false);
     let outlives_env = OutlivesEnvironment::with_bounds(param_env, implied_bounds);
 
+    check_for_redundant_lifetimes(tcx, body_def_id, &outlives_env);
```

Comment:
```suggestion
    lint_redundant_lifetimes(tcx, body_def_id, &outlives_env);
```

make it clear that this is only for linting